invenio_app_rdm/config.py

```diff
@@ -655,7 +659,9 @@ def files_rest_permission_factory(obj, action):
 # ===============
 # See https://invenio-indexer.readthedocs.io/en/latest/configuration.html
 
-INDEXER_DEFAULT_INDEX = "rdmrecords-records-record-v6.0.0"
+INDEXER_DEFAULT_INDEX = LocalProxy(
+    lambda: current_rdm_records_service.record_cls.index._name
+)
```

Comment:
In contrast to the cherry-picked PR which had `INDEXER_DEFAULT_INDEX = None`, I've used the same logic as above (just in case there's other code that relies on this in v12).